### PR TITLE
feat: integrate with gh

### PR DESCRIPTION
**Pull request by ArzelaAscoIi**
- add an action to publish the package on tag creation
- enables customers to use gh versioning & auto pushing to dC

### CODE REVIEW

File: .github/workflows/publish_on_tag.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Build and Push Custom Component
```
Comment (wochinge) [inline .github/workflows/publish_on_tag.yaml:1]: 🔥 Let's also add a testing step in the CI + documentation (e.g. how to trigger, where to put API key etc.)

**Comment by ArzelaAscoIi** [comment]
@wochinge works here: https://github.com/ArzelaAscoIi/dc-custom-component-template/actions/runs/11514208478/job/32052363295 

image
image

**Review by wochinge — COMMENTED** [review]
Spoke about this with Jul's yesterday @ArzelaAscoIi . It makes a lot of sense from our developer perspective but not so much from Sol / Customer perspective - I linked more information on the Jira item: https://deepset.atlassian.net/browse/DCD-468

We can still merge though - I just doubt it will see much use. Let's have Aga have a look over the docs first though.

File: .github/workflows/publish_on_tag.yaml (hunk 1 above)
Comment (ArzelaAscoIi) [reply .github/workflows/publish_on_tag.yaml:1]: let's do it in a separate PR

File: README.md [hunk 2]
```diff
@@ -96,3 +96,15 @@ It's crucial to thoroughly test your custom component before uploading it to dee
    This creates a zip file called `custom_component.zip` in the `dist` directory and uploads it to deepset Cloud.
 
 For detailed instructions, refer to our documentation on [Creating a Custom Component](https://docs.cloud.deepset.ai/docs/create-a-custom-component).
+
+### GitHub Actions
+
+We use GitHub Actions to build and push the custom component to deepset Cloud. When you create a tag, the build and push job will be triggered.
```
Comment (agnieszka-m) [inline README.md:102]:
```suggestion
We use GitHub Actions to build and push custom components to deepset Cloud. Create a tag to trigger the build and push job.
```

File: README.md [hunk 3]
```diff
@@ -96,3 +96,15 @@ It's crucial to thoroughly test your custom component before uploading it to dee
...
+After forking/cloning this repository, you need to
```
Comment (agnieszka-m) [inline README.md:103]:
```suggestion
After forking or cloning this repository:
```

File: README.md [hunk 4]
```diff
@@ -96,3 +96,15 @@ It's crucial to thoroughly test your custom component before uploading it to dee
    This creates a zip file called `custom_component.zip` in the `dist` directory and uploads it to deepset Cloud.
 
 For detailed instructions, refer to our documentation on [Creating a Custom Component](https://docs.cloud.deepset.ai/docs/create-a-custom-component).
+
+### GitHub Actions
+
+We use GitHub Actions to build and push the custom component to deepset Cloud. When you create a tag, the build and push job will be triggered.
+After forking/cloning this repository, you need to 
+
+1. add the following [secrets to your repository](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions):
```
Comment (agnieszka-m) [inline README.md:105]:
```suggestion
1. Add the `DEEPSET_CLOUD_API_KEY` [secret to your repository](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions). This is your deepset Cloud API key.
(To add a secret, go to your repository and choose _Settings > Secrets and variables > Actions > New repository secret_.)
```

File: README.md [hunk 5]
```diff
@@ -96,3 +96,15 @@ It's crucial to thoroughly test your custom component before uploading it to dee
...
+   - `DEEPSET_CLOUD_API_KEY`: Your deepset Cloud API key (repo > settings > secrets and variables > actions > new repository secret)
```
Comment (agnieszka-m) [inline README.md:106]:
```suggestion
```

**Review by agnieszka-m — COMMENTED** [review]
A couple of suggestions

**Pull request merged**